### PR TITLE
feat(wren-ai-service): optimize qdrant for multi users setting

### PR DESCRIPTION
**Pull request by cyyeh**
- add [binary quantization](https://qdrant.tech/documentation/guides/quantization/?q=binary)
- allow [multitenancy](https://qdrant.tech/documentation/guides/multiple-partitions/)
- offload vectors to disk

### CODE REVIEW

File: wren-ai-service/src/providers/document_store/qdrant.py [hunk 1]
```diff
@@ -291,9 +355,26 @@ def get_store(
...
             index=dataset_name or "Document",
             recreate_index=recreate_index,
+            on_disk=True,
+            quantization_config=(
+                rest.BinaryQuantization(
+                    binary=rest.BinaryQuantizationConfig(
+                        always_ram=True,
+                    )
+                )
+                if embedding_model_dim >= 1024
```
Comment (paopa) [inline wren-ai-service/src/providers/document_store/qdrant.py:369]: I think we can mention why the setting enable if the dimension more than 1024.

File: wren-ai-service/src/pipelines/indexing/indexing.py [hunk 2]
```diff
@@ -452,6 +464,7 @@ def visualize(self, mdl_str: str) -> None:
             output_file_path=f"{destination}/indexing.dot",
             inputs={
                 "mdl_str": mdl_str,
+                "id": id or "",
```
Comment (paopa) [inline wren-ai-service/src/pipelines/indexing/indexing.py:467]: I'm curious what will it happen if we don't handle with the id is None.

**Review by paopa — COMMENTED** [review]
overall LGTM. only have two questions for it.

File: wren-ai-service/src/pipelines/indexing/indexing.py (hunk 2 above)
Comment (cyyeh) [reply wren-ai-service/src/pipelines/indexing/indexing.py:469]: I've refactored the code

Comment (paopa) [reply wren-ai-service/src/pipelines/indexing/indexing.py:469]: I noticed that all the `id` parameters have been handled properly. I think we can also remove the empty string.

**Review by paopa — APPROVED** [review]
LGTM👍

**Pull request merged**